### PR TITLE
Mc userstory 2

### PR DESCRIPTION
**Pull request by Mariana-21**
**ALWAYS PULL FROM MASTER BEFORE CREATING PR**

### All Submissions:

* [X] Have you followed the guidelines in our Contributing document? (need to decided as group what to put here)
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls)? (for reducing merge conflicts)

### New Feature Submissions:

* [X] Have you added an explanation of what your changes do?
* [X] Have you lint your code locally prior to submission?

### Testing:

* [X] Does your submission pass tests?
* [X] Have you written new tests for your core changes, as applicable?
* [X] Have you successfully ran tests with your changes locally?

**Test coverage** = 100%
--------------------------------------------------------------------------------
Created a new test to ensure that all links are in the nav bar and all had links to respective pages. 
Created new route for new welcome page as well for the new login page.
Added links to the application html to ensure they were always on the nav bar.

### CODE REVIEW

File: spec/features/welcome/index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+        expect(current_path).to eq('/register')
+
+        within 'nav' do
+            click_link "Cart"
+        end
+
+        expect(current_path).to eq('/cart')
+    end
+end
```
Comment (tylertomlinson) [inline spec/features/welcome/index_spec.rb:44]: This test already exists as `navigation_spec.rb`. Would just need to add lines 26-36

File: app/controllers/welcome_controller.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class WelcomeController < ApplicationController
+
+    def index
```
Comment (tylertomlinson) [inline app/controllers/welcome_controller.rb:3]: I don't think we need an index method on the welcome controller

Comment (tylertomlinson) [inline app/controllers/welcome_controller.rb:3]: We don't need a index method on the welcome controller

File: app/controllers/login_controller.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class LoginController < ApplicationController
```
Comment (tylertomlinson) [inline app/controllers/login_controller.rb]: This will be `SessionsController`

File: config/routes.rb [hunk 4]
```diff
@@ -1,6 +1,9 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
 
+  get "/", to: "welcome#index"
+  get "/login", to: "login#index"
```
Comment (tylertomlinson) [inline config/routes.rb:5]: So the route will look like `/login, to #sessions#new`

Comment (tylertomlinson) [inline config/routes.rb]: So the route will look like `"/login, to #sessions#new"`

File: spec/features/welcome/index_spec.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+            click_link "Login"
+        end
+
+        expect(current_path).to eq('/login')
+
+        within 'nav' do
+            click_link "Register"
+        end
+
+        expect(current_path).to eq('/register')
```
Comment (tylertomlinson) [inline spec/features/welcome/index_spec.rb]: Nice work here! You can just add this (lines 26-36) to the existing navbar test under `navigation_spec.rb`

File: config/routes.rb (hunk 4 above)
Comment (danielpselph) [reply config/routes.rb]: Agree as well, I will be using these sessions for the login/out user stories.

**Review by danielpselph — CHANGES_REQUESTED** [review]
Updates needed for login controller to sessions controller

File: spec/features/welcome/index_spec.rb (hunk 5 above)
Comment (Mariana-21) [reply spec/features/welcome/index_spec.rb]: Yeah, I was thinking about that as well, so I will go ahead and do that now

File: spec/features/welcome/index_spec.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
+require 'rails_helper'
+
+describe "As a visitor", type: :feature do
+    it "can see a nav bar with link to the welcome page, all items, merchants, shopping carts, login/registration" do
```
Comment (alex-latham) [inline spec/features/welcome/index_spec.rb:4]: I think this test is missing some things described in the 'it' line

**Review by danielpselph — APPROVED** [review]

**Review by alex-latham — APPROVED** [review]
Looks good to me!

**Pull request merged**